hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java

```diff
@@ -314,15 +314,17 @@ private XceiverClientReply sendCommandWithRetry(
           // Pull the Cached DN to the top of the DN list
           Collections.swap(datanodeList, 0, getBlockDNCacheIndex);
         }
-      } else if (topologyAwareRead) {
```

Comment:
Good catch @sodonnel . The fix LGTM overall. I just have one question:

Have you consider populating blockID properly for the first getBlock() call with additional else if on the top?

else if (request.getCmdType() == ContainerProtos.Type.GetBlock) {
      blockID = request.getGetBlock().getBlockID();
    }

Reply:
Yea, I did wonder why getBlock() is not part of that IF block. I guess it makes sense to add it in just in case, as it will do no harm.

I am not really fully up-to-speed on the read path from the Ozone client - would it be normal for several getblock calls to be made against the same pipeline (XceiverClientGrpc instance)? Ie do we expect getBlock to benefit from the cached connection sometimes?

Reply:
> Yea, I did wonder why getBlock() is not part of that IF block. I guess it makes sense to add it in just in case, as it will do no harm.
> 
> I am not really fully up-to-speed on the read path from the Ozone client - would it be normal for several getblock calls to be made against the same pipeline (XceiverClientGrpc instance)? Ie do we expect getBlock to benefit from the cached connection sometimes?

Yes, it is quite possible to use the same XceiverClientGrpc instanvce to read for multiple blocks for multiple keys.

Reply:
Thanks for the feedback. I have made this change, so pending a clean CI run please give it a check.

Reply:
@xiaoyuyao @bshashikant Are you happy with the revision and for me to commit this one? Thanks.

Reply:
LGTM, +1.